Modules/binascii.c

```diff
@@ -592,6 +692,511 @@ binascii_b2a_base64_impl(PyObject *module, Py_buffer *data, int newline)
+    if (bin_data == NULL) {
+        return NULL;
+    }
+
+    /* Build ignore map. */
+    unsigned char ignore_map[256] = {0};
+    if (ignore->obj != NULL) {
+        const unsigned char *ignore_data = ignore->buf;
+        Py_ssize_t ignore_len = ignore->len;
+        for (Py_ssize_t i = 0; i < ignore_len; i++) {
+            this_ch = ignore_data[i];
+            ignore_map[this_ch] = -1;
+        }
+    }
+
+    uint32_t leftchar = 0;
+    int group_pos = 0;
+    for (; ascii_len > 0 || group_pos != 0; ascii_len--, ascii_data++) {
+        /* Shift (in radix-85) data or padding into our buffer. */
+        unsigned char this_digit;
+        if (ascii_len > 0) {
+            this_ch = *ascii_data;
+            this_digit = table_a2b_base85_a85[this_ch];
+        }
+        else {
+            /* Pad with largest radix-85 digit when decoding. */
+            this_digit = 84;
+        }
+        if (this_digit < 85) {
+            if (leftchar > UINT32_MAX / 85
+                || (leftchar *= 85) > UINT32_MAX - this_digit)
+            {
+                state = get_binascii_state(module);
+                if (state != NULL) {
+                    PyErr_SetString(state->Error, "Ascii85 overflow");
+                }
+                goto error;
+            }
+            leftchar += this_digit;
+            group_pos++;
+        }
+        else if ((this_ch == 'y' && fold_spaces) || this_ch == 'z') {
+            if (group_pos != 0) {
+                state = get_binascii_state(module);
+                if (state != NULL) {
+                    PyErr_Format(state->Error,
+                                 "'%c' inside Ascii85 5-tuple", this_ch);
+                }
+                goto error;
+            }
+            leftchar = this_ch == 'y' ? BASE85_A85_Y : BASE85_A85_Z;
+            group_pos = 5;
+        }
+        else if (!ignore_map[this_ch]) {
+            state = get_binascii_state(module);
+            if (state != NULL) {
+                PyErr_Format(state->Error,
+                             "Non-Ascii85 digit found: %c", this_ch);
+            }
+            goto error;
+        }
+
+        /* Wait until buffer is full. */
+        if (group_pos != 5) {
+            continue;
+        }
+
+        /* Write current chunk. */
+        Py_ssize_t chunk_len = ascii_len < 1 ? 3 + ascii_len : 4;
+        for (Py_ssize_t i = 0; i < chunk_len; i++) {
+            *bin_data++ = (leftchar >> (24 - 8 * i)) & 0xff;
+        }
+
+        group_pos = 0;
+        leftchar = 0;
+    }
+
+    return PyBytesWriter_FinishWithPointer(writer, bin_data);
+
+error:
+    PyBytesWriter_Discard(writer);
+    return NULL;
+}
+
+/*[clinic input]
+binascii.b2a_ascii85
+
+    data: Py_buffer
+    /
+    *
+    fold_spaces: bool = False
+        Emit 'y' as a short form encoding four spaces.
+    wrap: bool = False
+        Wrap result in '<~' and '~>' as in Adobe Ascii85.
+    width: unsigned_int(bitwise=True) = 0
```

Comment:
`bitwise=True` make it to wrap, so `2**32` is interpreted as 0.

There is also no benefit of restricting the range to the C's int, it have to support `sys.maxsize`. So use `size_t` here. We can also use a special converter, because the current Python code does not have limitation, and negative values interpreted as 1 (although this is an implementation detail, not documented and not tested, so it is not necessary to support this).